core-bundle/src/Picker/AbstractContentPickerProvider.php

```diff
@@ -0,0 +1,139 @@
+{
+    /**
+     * @var Security
+     */
+    private $security;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(FactoryInterface $menuFactory, RouterInterface $router, ?TranslatorInterface $translator, Security $security, Connection $connection)
+    {
+        parent::__construct($menuFactory, $router, $translator);
+
+        $this->security = $security;
+        $this->connection = $connection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsContext($context): bool
+    {
+        return 'content' === $context && $this->security->isGranted('contao_user.modules', $this->getBackendModule());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsValue(PickerConfig $config): bool
+    {
+        if (!is_numeric($config->getValue())) {
+            return false;
+        }
+
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('ptable')
+            ->from('tl_content')
+            ->where($qb->expr()->eq('id', $config->getValue()))
+        ;
+
+        if ($qb->execute()->fetchColumn() === $this->getParentTable()) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDcaTable(): string
+    {
+        return 'tl_content';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDcaAttributes(PickerConfig $config): array
+    {
+        $attributes = ['fieldType' => 'radio'];
+
+        $value = $config->getValue();
+        if ($fieldType = $config->getExtra('fieldType')) {
+            $attributes['fieldType'] = $fieldType;
+        }
+
+        if ($source = $config->getExtra('source')) {
+            $attributes['preserveRecord'] = $source;
+        }
+
+        if ($value) {
+            $intval = static function ($val) {
+                return (int) $val;
+            };
+
+            $attributes['value'] = array_map($intval, explode(',', $value));
+        }
+
+        return $attributes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function convertDcaValue(PickerConfig $config, $value)
+    {
+        return (int) $value;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getRouteParameters(PickerConfig $config = null): array
+    {
+        $params = ['do' => $this->getBackendModule()];
+
+        if (null !== $config) {
+            $qb = $this->connection->createQueryBuilder();
+            $qb->select('pid, ptable')->from('tl_content')->where($qb->expr()->eq('id', $config->getValue()));
+            $data = $qb->execute()->fetch();
+
+            if ($data['ptable'] === $this->getParentTable()) {
+                $params['table'] = 'tl_content';
+                $params['id'] = $data['pid'];
+            }
+        }
+
+        return $params;
+    }
+
```

Comment:
A short method description would be nice especially that you use `{@inheritdoc}` in child classes.

Reply:
Thanks, fixed in 1ce12911b7031b15d5436aa79e12b8dc87caedbe